calculating_n50_assemblies.py

```diff
@@ -0,0 +1,71 @@
+import re
+import gzip
+import operator
+import sys
+import os
+
+def main(in_dir = None, out_dir = None):
+    if os.path.exists(in_dir) == False: 
+        sys.exit("Input directory or path incorrect. Exiting..")
+    if os.path.exists(out_dir) == False: 
+        os.mkdir(out_dir)
+    global file_o
+    file_o = open(os.path.join(out_dir,'N50.txt'),'w')
+    fasta_ext = ['fasta','fna','ffn','faa','frn']
+
+    switch = 'off'
+    global f
+    for f_name in os.listdir(in_dir):
+        if f_name.endswith('.gz'):
+            array_f = f_name.split('.')
+            if array_f[-2] in fasta_ext:
+                file = os.path.join(in_dir,f_name)
+                with gzip.open(file, mode = 'rt') as f:
+                    n50_calc()
+                switch = 'on'
+        else:
+            array_f = f_name.split('.')
+            if array_f[-1] in fasta_ext:
+                file = os.path.join(in_dir,f_name)
+                with open(file, mode = 'r') as f:
+                    n50_calc()
+                switch = 'on'
+
+    if switch == 'off':
+        sys.exit("No zipped or unzipped fasta assemblies are in the directory. Exiting..")
+
+def n50_calc():
```

Comment:
I like the refactor, but I would avoid using global variables. Typically we reserve global variables for use as constants. Globals make it tricky for someone to debug your code and they could cause issues if you switch to multi-threading in the future.

Instead of using globals you can pass those file handles to the n50_calc method. By that I mean make those as parameters to the n50_calc method.

Reply:
Used def n50_calc(y = None): and passed a parameter n50_calc(f) to main.

Reply:
Could you please also pass the output file handle to the function, and if you could use more descriptive variable names that would also be great. Right now if some looks at your function and sees `y` they won't immediately know what it is.

Reply:
I replaced  file handle with open_file within main and also within the function. 

with gzip.open(file, mode = 'rt') as open_file:
                    n50_calc(open_file)

def n50_calc(open_file = None):
                     for line in open_file: 
                                  do_stuff

Is that it or am I overlooking something?

Reply:
You overlooked where you write to the output file.

Reply:
Right. Another "global" was sitting there. got rid of it and passed output file as parameter.